sysbuild/CMakeLists.txt

```diff
@@ -105,6 +109,10 @@ function(${SYSBUILD_CURRENT_MODULE_NAME}_post_cmake)
     include_provision_hex()
   endif()
 
+  if(DEFINED FP_MODEL_ID AND DEFINED FP_ANTI_SPOOFING_KEY)
+    include_fast_pair_hex()
```

Comment:
There is no `CONFIG_` here, sysbuild is not child/parent image, so there is nothing to check

Reply:
So maybe `subsys/bluetooth/services/fast_pair/CMakeLists.txt` we could execute this part:
```
if(NOT DEFINED FP_MODEL_ID OR NOT DEFINED FP_ANTI_SPOOFING_KEY)
    message(FATAL_ERROR "FP_MODEL_ID and FP_ANTI_SPOOFING_KEY build variables must be specified.")
  endif()
```
regardless of `SYSBUILD`? Currently it is hidden under `if(NOT SYSBUILD)`. That way we could fail build if Fast Pair and Sysbuild are used and FP_MODEL_ID or FP_ANTI_SPOOFING_KEY is not defined.

Reply:
No, the variables are present in the sysbuild context which then generates the hex file, they are not present in the image context

Reply:
Hmm I've also tried creating new file: `subsys/bluetooth/services/fast_pair/sysbuild.cmake` with following content:
```
if(NOT DEFINED FP_MODEL_ID OR NOT DEFINED FP_ANTI_SPOOFING_KEY)
  message(FATAL_ERROR "FP_MODEL_ID and FP_ANTI_SPOOFING_KEY build variables must be specified.")
endif()
```
but this file seems to never be executed. I don't see any other way to fail the build as we would want to.

Reply:
Also `nrf_desktop` if build with Fast Pair but without keys (`west build --sysbuild -b nrf52840dk_nrf52840 --pristine -- -DCONF_FILE=prj_fast_pair.conf`) fails weirdly:
![image](https://github.com/nrfconnect/sdk-nrf/assets/86837740/ff38a823-01c9-4ab5-a3a3-4579e82e9fed)
Instead it would be nice if it returned error related to missing `FP_MODEL_ID` and `FP_ANTI_SPOOFING_KEY`.

Reply:
@MarekPieta what do you think?

Reply:
> Did you cherry-pick these commits over to the sysbuild one? Because if not then nothing is going to build

Yes. This command is working fine for `nrf_desktop` app: `west build -b nrf52840dk_nrf52840 --pristine -- -DFP_MODEL_ID=0x8E717D -DFP_ANTI_SPOOFING_KEY=dZxFzP7X9CcfLPC0apyRkmgsh3n2EbWo9NFNXfVuxAM= -DCONF_FILE=prj_fast_pair.conf`, but the one above is failing in a strange way because of lack of model ID and Anti Spoofing Key.

Reply:
I don't know what you have done but it is not failing here, the sysbuild.cmake file sets the keys, then sysbuild generates a provision hex file. The above errors look like PM errors because you're not running the sysbuild-ncs PM code and the only reason for that would be you're not combing both PRs

Reply:
I checked the build issue observed by @alstrzebonski.
It seems that `CONFIG_BOOTLOADER_MCUBOOT` is no longer set in the application configuration even if application uses MCUboot bootloader. Is this expected? How should application check if bootloader is used with sysbuild?

Reply:
To clarify - the build failure can be observed in `applications/nrf_desktop` app when its built with Fast Pair support: `west build --sysbuild -b nrf52840dk_nrf52840 --pristine -- -DFP_MODEL_ID=0x8E717D -DFP_ANTI_SPOOFING_KEY=dZxFzP7X9CcfLPC0apyRkmgsh3n2EbWo9NFNXfVuxAM= -DCONF_FILE=prj_fast_pair.conf`

Reply:
> That will not be set if mcuboot is not enabled, and that application does not enable mcuboot, so I'm not sure why you would be asking about mcuboot in the context of this sample?

nRF Desktop application integrates MCUboot and enables it for Fast Pair configurations. The `dfu.c` application module checks which bootloader is in use to rely on proper Partition Manager symbols. It also requests image swap and confirms image if MCUboot bootloader is used.
Still, not relevant in context of Fast Pair sample.